mk/main.mk

```diff
@@ -72,6 +49,38 @@ CFG_RELEASE=$(CFG_RELEASE_NUM)-dev
 CFG_PACKAGE_VERS=$(CFG_RELEASE_NUM)-dev
 endif
 
+# Append a version-dependent hash to each library, so we can install different
+# versions in the same place
+CFG_FILENAME_EXTRA=$(shell printf '%s' $(CFG_RELEASE)$(CFG_EXTRA_FILENAME) | $(CFG_HASH_COMMAND))
+
+# A magic value that allows the compiler to use unstable features during the
+# bootstrap even when doing so would normally be an error because of feature
+# staging or because the build turns on warnings-as-errors and unstable features
+# default to warnings. The build has to match this key in an env var.
+#
+# This value is keyed off the release to ensure that all compilers for one
+# particular release have the same bootstrap key. Note that this is
+# intentionally not "secure" by any definition, this is largely just a deterrent
+# from users enabling unstable features on the stable compiler.
+CFG_BOOTSTRAP_KEY=$(CFG_FILENAME_EXTRA)
+
+# If local-rust is the same as the current version, then force a local-rebuild
+ifdef CFG_ENABLE_LOCAL_RUST
+ifeq (rustc $(CFG_RELEASE),\
+      $(shell $(Q)$(S)src/etc/local_stage0.sh --print-rustc-version $(CFG_LOCAL_RUST_ROOT)))
```

Comment:
OK, adding on to my note about `--version`, I see you are actually matching the "rustc" prefix at least.  The optional git info would throw it off.  I still think we should be a little more precise, because we might want vendor info to show in the basic `--version` string too, the same way gcc calls itself "`gcc version 6.1.1 20160621 (Red Hat 6.1.1-3) (GCC)`".